auth-web/src/components/auth/manage-business/AffiliationAction.vue

```diff
@@ -323,23 +332,34 @@ export default defineComponent({
       }
     }
 
-    const actionButtonText = (business: Business) => {
-      const affiliationInviteInfo = business.affiliationInvites[0]
-      const invitationStatus = affiliationInviteInfo.status
-      const statusActions = {
-        [AffiliationInvitationStatus.Pending]: 'Cancel<br>Request',
-        [AffiliationInvitationStatus.Failed]: 'Remove<br>from list',
-        [AffiliationInvitationStatus.Accepted]: 'Open'
-      }
-      return statusActions[invitationStatus] || ''
-    }
-
     // Actions
     const getPrimaryAction = (item: Business): string => {
-      const invitationStatus = item?.affiliationInvites?.[0]?.status
-      if ([AffiliationInvitationStatus.Pending, AffiliationInvitationStatus.Expired].includes(invitationStatus)) {
-        return 'Resend Email'
+      const affiliationInviteInfo = item?.affiliationInvites?.[0]
+      if ([AffiliationInvitationStatus.Pending,
+        AffiliationInvitationStatus.Expired,
+        AffiliationInvitationStatus.Failed].includes(affiliationInviteInfo?.status)) {
+        // checks for affiliation invitation
+        switch (true) {
+          case affiliationInviteInfo.type === AffiliationInvitationType.EMAIL &&
+          [AffiliationInvitationStatus.Pending, AffiliationInvitationStatus.Expired].includes(affiliationInviteInfo?.status):
+            return 'Resend Email'
+
+          case affiliationInviteInfo.type === AffiliationInvitationType.REQUEST &&
+          AffiliationInvitationStatus.Pending === affiliationInviteInfo?.status &&
+          isCurrentOrganization(item.affiliationInvites[0].fromOrg.id) :
+            return 'Cancel Request' // 'Cancel<br>Request'
```

Comment:
Does this need to be consistent with `showAffiliationInvitationCancelRequestButton` ?